tests/classy_integration/Controller_Test.php

```diff
@@ -2,85 +2,229 @@
+		$this->assertEquals( 'tec_common_assets_loaded', $added_actions[0]['hook'] );
+		$this->assertEquals( [ $controller, 'register_assets' ], $added_actions[0]['callback'] );
+		$this->assertEquals( 'tec_events_pro_classy_registered', $added_actions[1]['hook'] );
+		$this->assertEquals( [ $controller, 'register_ecp_editor_meta' ], $added_actions[1]['callback'] );
+	}
+
+	/**
+	 * @covers Controller::register_assets
+	 */
+	public function test_register_assets_registers_script_and_styles(): void {
 		/** @var Controller $controller */
 		$controller = $this->make_controller();
 
-		$data = $controller->filter_data( [] );
+		// Filter the supported post types.
+		add_filter(
+			'tec_classy_post_types',
+			static fn( $post_types ) => array_unique( array_merge( $post_types, [ TEC::POSTTYPE ] ) )
+		);
 
-		$this->assertEquals(
+		// Set up the global $post object to simulate a post using Classy.
+		global $post;
+		$post = $this->factory()->post->create_and_get(
 			[
-				'settings'       =>
-					[
-						'defaultCurrency'    =>
-							[
-								'code'     => 'USD',
-								'symbol'   => '$',
-								'position' => 'prefix',
-							],
-						'timeRangeSeparator' => ' - ',
-						'venuesLimit'        => 1,
-					],
-				'endOfDayCutoff' =>
-					[
-						'hours'   => 0,
-						'minutes' => 0,
-					],
-			],
-			$data
+				'post_type'   => TEC::POSTTYPE,
+				'post_status' => 'publish',
+			]
 		);
 
-		// Update the options.
-		tribe_update_option( 'defaultCurrencyCode', 'EUR' );
-		tribe_update_option( 'defaultCurrencySymbol', '€' );
-		tribe_update_option( 'reverseCurrencyPosition', 'postfix' );
-		tribe_update_option( 'timeRangeSeparator', ' ~ ' );
-		tribe_update_option( 'multiDayCutoff', '02:30' );
+		// Mock Asset::add to capture calls.
+		$asset_calls = [];
+		$this->mock_assets_add( $asset_calls );
 
-		$data = $controller->filter_data( [] );
+		// Verify both script and styles were registered.
+		$controller->register_assets();
+		$this->assertCount( 2, $asset_calls );
+		$this->assertEquals( 'tec-classy-tickets', $asset_calls[0]['handle'] );
+		$this->assertEquals( 'classy.js', $asset_calls[0]['file'] );
+		$this->assertEquals( 'tec-classy-tickets-styles', $asset_calls[1]['handle'] );
+		$this->assertEquals( 'style-classy.css', $asset_calls[1]['file'] );
+	}
 
-		$this->assertEquals(
-			[
-				'settings'       =>
-					[
-						'defaultCurrency'    =>
-							[
-								'code'     => 'EUR',
-								'symbol'   => '€',
-								'position' => 'postfix',
-							],
-						'timeRangeSeparator' => ' ~ ',
-						'venuesLimit'        => 1,
-					],
-				'endOfDayCutoff' =>
-					[
-						'hours'   => 2,
-						'minutes' => 30,
-					],
-			],
-			$data
+	/**
+	 * @covers Controller::get_et_class
+	 */
+	public function test_get_et_class_returns_et_class_name(): void {
+		/** @var Controller $controller */
+		$controller = $this->make_controller();
+
+		// Use reflection to access private method.
+		$method = Closure::bind(
```

Comment:
If it's private it should not be tested. You can remove this test and cover the method implicitly in other tests.